linker/shared/src/main/scala/org/scalajs/linker/frontend/IRLoader.scala

```diff
@@ -121,6 +131,13 @@ private final class ClassDefAndInfoCache {
           version.get != newVersion.get) {
         version = newVersion
```

Comment:
Should we move this assignment after the one to `cacheUpdate`, now that the latter can "likely" throw?

I don't think the `IRLoader` is protected by the "any exception invalidates the whole state" rule, is it?

Reply:
The assignment to `cacheUpdate` will not throw. The checking is asynchronous: So the `Future` representing the load will be in a failed state. IMO this is consistent: A new version of the file could be a non failed state (although I don't think the linker is able to use this).